### ISSUE TITLE
 DeprecationWarning: 'U' mode is deprecated

### ISSUE THREAD

**Issue by zopyx** [zopefoundation/Zope#415]
```
ajung@dev ~/src/pp.connector.plone (master) $ bin/wsgi.py fg

/home/ajung/.buildout/eggs/Zope-4.0b7-py3.7.egg/Zope2/Startup/serve.py:273: DeprecationWarning: 'U' mode is deprecated
```

**Comment by icemac** [comment]
Hm, https://github.com/zopefoundation/Zope/blob/master/src/Zope2/Startup/serve.py#L219 ends at line 219 What's on your line 273?

**Comment by zopyx** [comment]
same here...the error message is misleading or at least somehow shadowing the right location

**Comment by icemac** [comment]
@zopyx Could you please grep the code in your instance for `'U'` resp `"U"` where it does come from?

**Comment by sallner** [comment]
There is an open PR in https://github.com/buildout/buildout/pull/476

**Comment by dataflake** [comment]
@sallner If this is proven to be a buildout issue then you should close it here.

**Comment by icemac** [comment]
This issue here does not seem to be related to the issue in buildout as it happens during running Zope.

**Comment by dataflake** [comment]
I'm wondering what that `bin/wsgi.py` is that @zopyx us using and what's creating it.  There's still no clear picture whether this is a Zope issue at all.

**Comment by zopyx** [comment]
I used two parts `instance` and `wsgi.py` for testing the new WSGI support for a while. So the configuration with `wsgi = on` generated the `bin/wsgi.py` control script.

**Comment by icemac** [comment]
In union.cms (also built using `plone.recipe.zope2instance`) I also see this message when starting an instance in the foreground: 
`/home/gocept/.buildout/eggs/Zope-4.0-py3.7.egg/Zope2/Startup/serve.py:288: DeprecationWarning: 'U' mode is deprecated`

Only the line number differs.

**Comment by icemac** [comment]
I tracked it down to the following:

Starting the instance results in a `subprocess` call with the following arguments: `/home/gocept/bo_verdi_intra_clients_dev/default/bin/python /home/gocept/bo_verdi_intra_clients_dev/default/parts/zcl1/bin/interpreter /home/gocept/.buildout/eggs/Zope-4.0-py3.7.egg/Zope2/Startup/serve.py /home/gocept/bo_verdi_intra_clients_dev/default/parts/zcl1/etc/wsgi.ini -d debug-mode=on`

The used `bin/interpreter` script has in its line 288: `with open(__file__, 'U') as __file__f:`.

**Comment by icemac** [comment]
`bin/interpreter` is generated via https://github.com/buildout/buildout/blob/c8f0236256cb8790db96764aad5754d93f2204d9/src/zc/buildout/easy_install.py#L1499 and should be fixed via buildout/buildout#476

This means it can be closed here.
